ui/webpack.config.js

```diff
@@ -1,22 +1,64 @@
-module.exports = {
-    context: __dirname,
-    entry: "./app/components/Main.js",
-    output: {
-        path: __dirname + "/client-react/js",
-        filename: "bundle.js"
-    },
-    module: {
+var webpack = require('webpack');
+var HtmlWebpackPlugin = require('html-webpack-plugin');
+var inDev = process.env.NODE_ENV = 'development';
+
+module.exports = (function() {
+    var config = {
+        'modulesDirectories': ['node_modules']
+    };
+
+    // Define entry points
+    config.entry = {
+        'app': __dirname + '/app/src/app.js'
+    };
+
+    config.output = {
+        'path': __dirname + '/app/dist/',
```

Comment:
I don't like the idea to have the dist folder inside the src folder and the folder we serve the dev environment from. This should be on the same level as app/. I would like to create a build folder in ui and create the dist inside that. Something like: /build/dist/
But maybe I don't understand the proposed structure and you need to explain it more.